addons/sale_timesheet/models/product.py

```diff
@@ -68,6 +68,17 @@ def _compute_product_tooltip(self):
                         "project for the order to track the time spent."
                     )
 
+    @api.onchange('type', 'service_type', 'service_policy')
+    def _update_uom_id(self):
+        for record in self:
+            if record.type == 'service' and record.service_type == 'timesheet' and \
+               not (record._origin.service_policy and record.service_policy == record._origin.service_policy):
+                record.uom_id = self.env.ref('uom.product_uom_hour')
+            elif record._origin.uom_id:
+                record.uom_id = record._origin.uom_id
+            else:
+                record.uom_id = self._get_default_uom_id()
```

Comment:
> Why did you choose to do an onchange instead of a compute method? monocle_face

@xavierbol Thank you again for the review :pray: To get back on this: I chose to use an onchange method to avoid undesired behaviors that could have happened imo with a compute method with a depends decorator.

The way I understand the spec, this change is more about suggesting a default time uom to the user when he selects a service product, rather than setting this blindly in every circumstances. For example, if the service type of a record is changed anywhere using the orm, I think it is preferable to not change the uom than setting it arbitrary to hour, without even changing the value of the price of the product.

In summary, I don't really see a use case where we would prefer a compute method to this onchange implementation. Let me know your opinion about this.

Reply:
Indeed, we discussed about it and indeed we will keep the onchange, however, we have to call it `_onchange_uom_id` (maybe that onchange already exists in sale module)

Also, the onchange has to be defined in `product.template` as well, to be able to have the same behavior in that model since `product.product` is a `product.template` with additional fields and features (thanks to `inherits`).

Reply:
> Indeed, we discussed about it and indeed we will keep the onchange, however, we have to call it `_onchange_uom_id` (maybe that onchange already exists in sale module)
> 
> Also, the onchange has to be defined in `product.template` as well, to be able to have the same behavior in that model since `product.product` is a `product.template` with additional fields and features (thanks to `inherits`).

I've changed the onchange method as discussed and added some tests. Let me know if you have other comments.